eth2util/eth2exp/attagg_test.go

```diff
@@ -0,0 +1,149 @@
+
+	_, secret, err := tbls.KeygenWithSeed(rand.New(rand.NewSource(1)))
+	require.NoError(t, err)
+
+	sigRoot, err := eth2util.SlotHashRoot(slot)
+	require.NoError(t, err)
+
+	slotsPerEpoch, err := bmock.SlotsPerEpoch(context.Background())
+	require.NoError(t, err)
+
+	epoch := eth2p0.Epoch(uint64(slot) / slotsPerEpoch)
+	sigData, err := signing.GetDataRoot(context.Background(), bmock, signing.DomainSelectionProof, epoch, sigRoot)
+	require.NoError(t, err)
+
+	sig, _ := tbls.Sign(secret, sigData[:])
+	blssig := tblsconv.SigToETH2(sig)
+
+	t.Run("is aggregator", func(t *testing.T) {
+		commLen := 43
+		var vals []eth2p0.ValidatorIndex
+		for idx := 1; idx <= commLen; idx++ {
+			vals = append(vals, eth2p0.ValidatorIndex(idx))
+		}
+		comms := []*eth2v1.BeaconCommittee{
+			{
+				Slot:       slot,
+				Index:      commIdx,
+				Validators: vals,
+			},
+		}
+
+		bmock.BeaconCommitteesAtEpochFunc = func(_ context.Context, _ string, _ eth2p0.Epoch) ([]*eth2v1.BeaconCommittee, error) {
+			return comms, nil
+		}
+
+		subscription := eth2exp.BeaconCommitteeSubscription{
+			ValidatorIndex: eth2p0.ValidatorIndex(1),
+			Slot:           slot,
+			CommitteeIndex: commIdx,
+			SlotSignature:  blssig,
+		}
+
+		resp, err := eth2exp.CalculateCommitteeSubscriptionResponse(ctx, bmock, subscription)
+		require.NoError(t, err)
+		require.Equal(t, resp.ValidatorIndex, subscription.ValidatorIndex)
+		require.True(t, resp.IsAggregator)
+	})
+
+	t.Run("is not aggregator", func(t *testing.T) {
+		commLen := 61
+		var vals []eth2p0.ValidatorIndex
+		for idx := 1; idx <= commLen; idx++ {
+			vals = append(vals, eth2p0.ValidatorIndex(idx))
+		}
+		comms := []*eth2v1.BeaconCommittee{
+			{
+				Slot:       slot,
+				Index:      commIdx,
+				Validators: vals,
+			},
+		}
+
+		bmock.BeaconCommitteesAtEpochFunc = func(_ context.Context, _ string, _ eth2p0.Epoch) ([]*eth2v1.BeaconCommittee, error) {
+			return comms, nil
+		}
+
+		subscription := eth2exp.BeaconCommitteeSubscription{
+			ValidatorIndex: eth2p0.ValidatorIndex(1),
+			Slot:           slot,
+			CommitteeIndex: commIdx,
+			SlotSignature:  blssig,
+		}
+
+		resp, err := eth2exp.CalculateCommitteeSubscriptionResponse(ctx, bmock, subscription)
+		require.NoError(t, err)
+		require.Equal(t, resp.ValidatorIndex, subscription.ValidatorIndex)
+		require.False(t, resp.IsAggregator)
+	})
+}
+
+func TestIsAggregator(t *testing.T) {
+	bmock, err := beaconmock.New()
+	require.NoError(t, err)
+
+	sig, err := hex.DecodeString("8776a37d6802c4797d113169c5fcfda50e68a32058eb6356a6f00d06d7da64c841a00c7c38b9b94a204751eca53707bd03523ce4797827d9bacff116a6e776a20bbccff4b683bf5201b610797ed0502557a58a65c8395f8a1649b976c3112d15")
+	require.NoError(t, err)
+	blsSig, err := tblsconv.SigFromBytes(sig)
+	require.NoError(t, err)
+	eth2Sig := tblsconv.SigToETH2(blsSig)
+
+	t.Run("is aggregator", func(t *testing.T) {
+		commLen := 3
+		isAgg, err := eth2exp.IsAggregator(context.Background(), bmock, int64(commLen), eth2Sig)
```

Comment:
https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/core/helpers/attestation_test.go#L29

Reply:
add as comment in the code please